### ISSUE TITLE
Disabling VSync on MacOS doesn't work as expected

### ISSUE THREAD

**Issue by memmaker** [hajimehoshi/ebiten#2543]
### Ebitengine Version

v2.5.0-alpha.9.1

### Operating System

- [ ] Windows
- [X] macOS
- [ ] Linux
- [ ] FreeBSD
- [ ] OpenBSD
- [ ] Android
- [ ] iOS
- [ ] Nintendo Switch
- [ ] Xbox
- [ ] Web Browsers

### Go Version (`go version`)

go1.19.4 darwin/amd64

### What steps will reproduce the problem?

`ebiten.SetVsyncEnabled(false)`
`ebiten.RunGame(c)`

### What is the expected result?

The application window should redraw itself as fast as possible. FPS way above 60 for an idle game state like a menu are expected.

### What happens instead?

The FPS look like they are still locked to VSync most of the time, staying in the 60s. There are some random spikes bringing the FPS exactly to 120 FPS for a short time before going back down to 60. There is no apparent relationship between the load of the game and the FPS being shown.

### Anything else you feel useful to add?

While this was originally discovered in a project that uses `ebiten.SetScreenClearedEveryFrame(false)`, I checked and it also is the same behaviour with `ebiten.SetScreenClearedEveryFrame(true)`.

NOTE: I was using the "ebitenginesinglethread" tag on my Mac because it is twice as fast.

**Comment by memmaker** [comment]
Just did another test without the "ebitenginesinglethread" build tag:
The FPS are now fluctuating much more wildly between exactly 60 and exactly 120 FPS. Still no apparent relationship to the load of the application. Still not the FPS I'd expect in an idle state with VSync off.

**Comment by hajimehoshi** [comment]
I couldn't reproduce the issue.

I tried `examples/windowsize` with and without `ebitenginesinglethread`, and reached 180 FPS without vsync.

```
                    'c.          hajimehoshi@Hajimes-MacBook-Pro-2020.local 
                 ,xNMM.          ------------------------------------------ 
               .OMMMMo           OS: macOS 13.0 22A380 x86_64 
               OMMM0,            Host: MacBookPro16,2 
     .;loddo:' loolloddol;.      Kernel: 22.1.0 
   cKMMMMMMMMMMNWMMMMMMMMMM0:    Uptime: 8 days, 19 hours, 3 mins 
 .KMMMMMMMMMMMMMMMMMMMMMMMWd.    Packages: 188 (brew) 
 XMMMMMMMMMMMMMMMMMMMMMMMX.      Shell: bash 3.2.57 
;MMMMMMMMMMMMMMMMMMMMMMMM:       Resolution: 1680x1050@2x 
:MMMMMMMMMMMMMMMMMMMMMMMM:       DE: Aqua 
.MMMMMMMMMMMMMMMMMMMMMMMMX.      WM: Quartz Compositor 
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.    WM Theme: Blue (Light) 
 .XMMMMMMMMMMMMMMMMMMMMMMMMMMk   Terminal: tmux 
  .XMMMMMMMMMMMMMMMMMMMMMMMMK.   CPU: Intel i7-1068NG7 (8) @ 2.30GHz 
    kMMMMMMMMMMMMMMMMMMMMMMd     GPU: Intel Iris Plus Graphics 
     ;KMMMMMMMWXXWMMMMMMMk.      Memory: 23585MiB / 32768MiB 
       .cooc,.    .,coo:.
                                                         
                                                         
```

**Comment by memmaker** [comment]
That would still feel wrong to me if I was you.

1) I've been told an idle game without VSync should reach 800+ FPS. So should the example. Also 180 is exactly 60*3, this is strange. It feels limited.

2) I would expect numbers much higher than 180 and also numbers that aren't exactly a multiple of 60.

**Comment by memmaker** [comment]
I think you did reproduce it actually, 180FPS is not what I would expect.

**Comment by hajimehoshi** [comment]
What if you use OpenGL instead of Metal?

**Comment by memmaker** [comment]
Yes, that does what I would expect. It will run at about 400+FPS then. So it is a metal thing.
